crates/sui-indexer/migrations/pg/2024-09-12-213234_watermarks/up.sql

```diff
@@ -0,0 +1,22 @@
+CREATE TABLE watermarks
+(
+    -- The table name of group of tables governed by this watermark, i.e `epochs`, `checkpoints`,
+    -- `transactions`.
+	entity                      TEXT          NOT NULL,
+    -- Upper bound epoch range to enable per-entity epoch-level retention policy.
+    epoch_hi                    BIGINT        NOT NULL,
+    -- Lower bound epoch range to enable per-entity epoch-level retention policy.
+    epoch_lo                    BIGINT        NOT NULL,
+    -- The inclusive high watermark that the committer advances.
+	hi                          BIGINT        NOT NULL,
```

Comment:
Is this in terms of checkpoint sequence number? Would be nice to comment

Reply:
+1 -- can we indicate it in the name as well?

Reply:
Also, indentation (I think you are using tabs instead of spaces...?)

Reply:
> (I think you are using tabs instead of spaces...?)

You trying to start a holy war?! ;P

Reply:
Yes an extra comment here would be good. I think the idea is that the high and low BIGINT may mean checkpoint in one case and txn seq number in other so it may be dependent on the table its tracking. but i'll let Will make that clarification

Reply:
> You trying to start a holy war?! ;P

😅 -- I guess the accurate framing is that we have a mix of tabs and spaces in this file.

Reply:
If it wasn't obvious I was trying to joke, 100% agree about being consistent with the formatting of our other migration files

Reply:
> may mean checkpoint in one case and txn seq number in other

Ah, interesting. This is kind of tangentially related but what do we use the epoch columns for on each watermark?

Reply:
that's a good question, i'm not sure if it provides much value right now as-is (but i may be missing something)

Reply:
Yes like @bmwill mentioned, the `hi` and `lo` mean different units per the table its tracking

> what do we use the epoch columns for on each watermark?
the objects history table has a separate retention policy of 2 epochs, so `epoch_lo` and `epoch_hi` allows `objects_history` and other tables to track their own retention independent of the `epochs` table